lib/instrumentation/index.js

```diff
@@ -152,6 +153,20 @@ Instrumentation.prototype.bindFunction = function (original) {
   }
 }
 
+Instrumentation.prototype.bindEmitter = function (emitter) {
+  var ins = this
+
+  shimmer.massWrap(emitter, [
+    'on',
```

Comment:
`on` points _directly_ to `addListener`, it doesn't _call_ it, therefore if we only patch one of the functions, the other would remain unwrapped.

As for `once` and `prependOnceListener`, those _do_ call the underlying `on` and `prependListener` functions, so wrapping those _would_ result in double-wrapping.

Reply:
> on points directly to addListener, it doesn't call it, therefore if we only patch one of the functions, the other would remain unwrapped.

Ah yes, I wasn't paying attention.

> As for once and prependOnceListener, those do call the underlying on and prependListener functions, so wrapping those would result in double-wrapping.

As mentioned in my previous comment, we have a guard to make sure double wrapping doesn't happen. So doing this would only be to ensure this would work even if Node core in the future didn't forward the call. But this is highly unlikely to happen, so I think we're safe not do it 👍 

But it might make sense to add a few quick tests to see that it works also for the aliases and the `once` versions.

Reply:
I've added a bunch more tests proving behaviour of both `bindFunction` and `bindEmitter`.